Final_Project_Paper-AR.qmd

```diff
@@ -0,0 +1,174 @@
+```
+In the Foreign Policy Bulletin Dataset, European nations and the Axis Powers are disproportionately focused on. In raw numbers, from 1936-1946, Britain, Germany, Russia, and Japan were directly mentioned 12,027 times. Meanwhile, Argentina, Brazil, Chile, Mexico, and Venezuela were only directly referenced 1,357 times. Yet, the bulletin’s lackluster coverage of Latin America in juxtaposition to its European counterparts should not diminish Latin America’s significance to U.S. foreign policy. The major theme associated with Latin American nations in the bulletin indicates the critical role that Latin America played in protecting U.S. interests. 
+
+```{r}
+#topic model
+tidy_fp_words <- tidy_fp %>% count(filename, word)
+
+fp.dtm <- tidy_fp_words %>% 
+  count(filename, word) %>% 
+  cast_dtm(filename, word, n)
+
+fp3.lda <- LDA(fp.dtm, k = 16, control = list(seed = 12345), method = "Gibbs", alpha = 0.5)
+fp3.lda
+
+fp3.topics <- tidy(fp3.lda, matrix = "beta")
+
+fp3.top.terms <- fp3.topics %>%
+  arrange(desc(beta)) %>% 
+  group_by(topic) %>% slice(1:10)
+
+fp3.top.terms %>%
+  mutate(term = reorder_within(term, beta, topic)) %>%
+  ggplot(aes(beta, term, fill = factor(topic))) +
+  geom_col(show.legend = FALSE) +
+  facet_wrap(~ topic, scales = "free") +
+  scale_y_reordered()
+```
+With regard to the topic model output that focuses on Latin American nations, the topic model found that Argentina, Brazil, Mexico, Chile, and solidarity were co-occurring. The topic model was picked up on a major foreign policy theme that was present throughout the interwar period and World War II, which is inter-American cooperation. Since neutrality sentiment was prevalent in the United States, Roosevelt was largely unable to act outside of the Americas until the attack on Pearl Harbor. Roosevelt, instead, chose to promote inter-American solidarity to prevent the Axis Powers from penetrating into the Western Hemisphere. This inter-American cooperation manifested through Roosevelt’s Good Neighbor diplomacy, but it still had its limitations. Inter-American cooperation did not mean that Latin American nations were required to make a direct alliance with the United States; rather, it was intended to create a common bond and galvanize support for keeping out the Axis Powers from the Western Hemisphere.^[fn5: Gellman, Good Neighbor Diplomacy, 12] Indeed, word vectors further suggest this connection between solidarity and the Western Hemisphere.
+
+```{r}
+#word vectors
+if (!require(wordVectors)) {
+  if (!(require(devtools))) {
+    install.packages("devtools")
+  }
+  devtools::install_github("bmschmidt/wordVectors")
+}
+
+if (!file.exists("txt")) prep_word2vec(origin="ForeignPolicyBulletinData", destination="txt", lowercase=T, bundle_ngrams=1)
+
+if (!file.exists("corpus_vectors.bin")) {model = train_word2vec("txt","corpus_vectors.bin",vectors=200, threads=5, window=12, iter=5, negative_samples=0)} else model = read.vectors("corpus_vectors.bin")
```

Comment:
Rather than running this inside your document, where it runs everytime it is being rendered - you could include it in your exploratory data analysis and then just load the vector model with ```model = read.vectors("corpus_vectors.bin")``` 

For some reason the .bin and .txt files that are necessary to run this code weren't included in your repo? Make sure those are included as they are necessary for others to run your model.